tests/kernel_bundle/get_kernel_bundle_builtin_kernels.cpp

```diff
@@ -0,0 +1,327 @@
+//    4) sycl::get_kernel_bundle<BundleState>(context, kernelIds)
+//    5) sycl::get_kernel_bundle<BundleState>(context, devices, kernelIds)
+//    6) sycl::get_kernel_bundle<BundleState>(context)
+//  In this tests uses two user defined kernels without any requirements without
+//  any requirements for device from chosen platform.
+//
+//  This test verifies that this overload:
+//    1) sycl::get_kernel_bundle<BundleState>(context)
+//    2) sycl::get_kernel_bundle<BundleState>(context, devices)
+//    3) sycl::get_kernel_bundle<BundleState>(context, devices, selector)
+//  Doesent contained built-in kernels.
+//  This test verifies that this overload doesent contained built-in kernels:
+//    1) sycl::get_kernel_bundle<BundleState>(context, selector)
+//    2) sycl::get_kernel_bundle<BundleState>(context, kernelIds)
+//    3) sycl::get_kernel_bundle<BundleState>(context, devices, kernelIds)
+//  Contains built-in kernels (if built-in kernel_ids were provided) and
+//  contains built-in kernels + user defined kernels (if built-in kernel_ids +
+//  user defined kernel_ids were provided).
+//
+*******************************************************************************/
+
+#include "../common/common.h"
+#include "get_kernel_bundle.h"
+#include "kernels.h"
+
+#define TEST_NAME get_kernel_bundle_builtin_kernels
+
+namespace TEST_NAMESPACE {
+using namespace sycl_cts;
+using namespace kernels;
+using namespace sycl_cts::tests::get_kernel_bundle;
+using namespace sycl_cts::tests::kernel_bundle;
+
+using first_simple_kernel = simple_kernel_descriptor::type;
+using second_simple_kernel = simple_kernel_descriptor_second::type;
+
+const auto first_simple_kernel_id = sycl::get_kernel_id<first_simple_kernel>();
+const auto second_simple_kernel_id =
+    sycl::get_kernel_id<second_simple_kernel>();
+const std::vector<sycl::kernel_id> user_defined_kernel_ids{
+    first_simple_kernel_id, second_simple_kernel_id};
+
+
+
+/** @brief Do some verifications for described sycl::get_kernel_bundle overloads
+ *  @tparam BundleState sycl::bundle_state enumeration's field
+ *  @param log sycl_cts::util::logger class object
+ *  @param queue sycl::queue class object
+ */
+template <sycl::bundle_state BundleState>
+void run_tests(util::logger &log, sycl::queue &queue) {
+  const auto context { queue.get_context()};
+  const auto device {queue.get_device()};
+  bool selector_was_called_for_built_in {false};
+
+  const std::vector<sycl::kernel_id> built_in_kernel_ids { device.template get_info<sycl::info::device::built_in_kernel_ids>()};
+
+  const auto selector_1 = [](const sycl::device_image<BundleState> &) {
+    return true;
+  };
+  const auto selector_2 =
+      [&](const sycl::device_image<BundleState> &device_img) {
+        for (auto &built_in_kernel_id : built_in_kernel_ids) {
+          if (!device_img.has_kernel(built_in_kernel_id)) {
+            selector_was_called_for_built_in = true;
+          }
+        }
```

Comment:
In this case using some alghorithm its not so easy, because of we should save `device_img` somewhere.

> Furthermore, as soon as selector_was_called_for_built_in is true, it is no longer required to be computed again in the next selector invocation, so you can guard this with an if (!selector_was_called_for_built_i).

I just break the sycle in case `!device_img.has_kernel(built_in_kernel_id)` is `true`.
I've done it [here](https://github.com/KhronosGroup/SYCL-CTS/pull/224/commits/c151a8cdae55ff2813aa9db1ffde34b33ebd1c11#diff-cfce662345b154bd62edfd6bfceb4dc9e924282661c3234f57122c0262abe876R85)